### DISCUSSION TITLE
Is it a bad practice to add custom messages in `postprocess` of processor?

### DISCUSSION THREAD

**Discussion by ylemkimon** [eslint/eslint#13583] [General] [answered]
I'm developing a plugin with a processor.

In the `preprocess` step, processing the text may result in warnings or errors.

Would it be a bad practice to add them as messages in the `postprocess` step?

Furthermore, would it be a bad practice to implement a linting rule using a processor, i.e., check the code in `preprocess` and print the message in `postprocess`? Implementing a custom parser and rule seems like overkill.

**Comment by kaicataldo** [comment, marked as answer]
If it's a error that occurs during parsing, I don't think it's a bad idea. A custom rule could cover it, too, but I think the former is more common.
